typescript/cli/src/utils/keys.ts

```diff
@@ -36,6 +36,8 @@ export async function getImpersonatedSigner({
   skipConfirmation?: boolean;
 }) {
   key ||= await retrieveKey(IMPERSONATED_KEY_TYPE, skipConfirmation);
+  if (key.length != ETHEREUM_ADDRESS_LENGTH)
+    key = privateKeyToSigner(key).address;
```

Comment:
don't know if this is your code initially - but it seems like we're overloading the key to be both a private key and an address here? again i like explicitness (privateKey vs address)

Reply:
this PR is the result of that conversation– where **`HYP_KEY` can now only be used to source a private key**.

In addition, a developer may provide a private key or an address to `--key` when running a dry-run (@yorhodes and I decided this would be useful for those dry-running for an arbitrary address, or not wanting to provide their PK just yet)

Reply:
I would suggest to go further with the separation
if a user wants to pass an address for dry run, they should pass an --address or --from flag
passing different types to the --key flag is still confusing